src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1480,14 +1480,10 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RequestSEDFastP
             mFastPollingConsumers--;
     }
 
-    if (mFastPollingConsumers > 0)
-    {
-        err = SetSEDPollingMode(ConnectivityManager::SEDPollingMode::Active);
-    }
-    else
-    {
-        err = SetSEDPollingMode(ConnectivityManager::SEDPollingMode::Idle);
-    }
+    mode = mFastPollingConsumers > 0 ? ConnectivityManager::SEDPollingMode::Active : ConnectivityManager::SEDPollingMode::Idle;
+
+    if (mPollingMode != mode)
+        err = SetSEDPollingMode(mode);
```

Comment:
SetSEDPollingMode already does this.  This also breaks _SetSEDPollingConfig, which is supposed to write through the polling intervals even if we haven't changed mPollingMode.

Reply:
> SetSEDPollingMode already does this

Yes I know that, but I thought it would be valuable to break earlier and to not execute few lines if it will not result in changing anything.

> This also breaks _SetSEDPollingConfig, which is supposed to write through the polling intervals even if we haven't changed mPollingMode.

Could you explain why is that, as I can't figure it out? `_SetSEDPollingConfig` sets new config values and calls `SetSEDPollingMode` with the current mode just to to set new poll period to the OpenThread. The case of calling `RequestSEDFastPollingMode` is to change the mode, but with no changes to the polling config values.

Reply:
@msandstedt ?

Reply:
Yes, but it does change things.  I'll read again, but I'm very sure this will break things.

Reply:
Ah, I see.  Yes, you are correct.  This works.  I still question the optimization because it's more complex and introduces more subtle statefulness.  But it doesn't break anything.